### PR TITLE
Bug fix/bts 1548

### PR DESCRIPTION
**Pull request by mchacki**
### Scope & Purpose

*Include a leading Collection in the new CollectionGroups. This is not required at all for Replication2 to work. However it is the easiest to make it a drop-in replacement for Replication 1 in the beginning. We reintroduce the limitation that we cannot drop a leading collection, but we gain that the behavior is the same, and we are protected against unexpected changes of "distributeShardsLike" responses. We may lift this restriction sometime later if we had time to adapt all our tools.*

One of the "tools" that used distributeShardsLike and relied on it to be a non-modifiable value is SmartGraphs, this caused: https://arangodb.atlassian.net/browse/BTS-1548

With this restriction test in above ticket is green again.

I did not add additional tests on purpose this behavior is covered by existing tests, which only run for replication 1.0 right now.

Side-Querst: This PR removes some functions that are not called anymore.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-1548
- [ ] Design document:

### CODE REVIEW

File: arangod/VocBase/LogicalCollection.cpp [hunk 1]
```diff
@@ -357,20 +357,24 @@ replication::Version LogicalCollection::replicationVersion() const noexcept {
...
-  TRI_ASSERT(_sharding != nullptr);
-  return _sharding->avoidServers();
+  if (ServerState::instance()->isCoordinator() &&
+      replicationVersion() == replication::Version::TWO) {
+    auto& cf = vocbase().server().getFeature<ClusterFeature>();
+    auto& ci = cf.clusterInfo();
+
+    auto myGroup = ci.getCollectionGroupById(groupID());
+    TRI_ASSERT(myGroup != nullptr)
+        << "Collection part of a Group that does not exist";
```
Comment (goedderz) [inline arangod/VocBase/LogicalCollection.cpp:367]: Can this be a race? E.g., can the LogicalCollection object still be around for a while after the collection group has already been deleted, and is thus missing from ClusterInfo?

**Review by goedderz — COMMENTED** [review]
Just one question, otherwise LGTM

File: arangod/VocBase/LogicalCollection.cpp (hunk 1 above)
Comment (mchacki) [reply arangod/VocBase/LogicalCollection.cpp:367]: Good point I think it actually can.
Will add graceful handling.

**Review by goedderz — APPROVED** [review]

**Comment by mchacki** [comment]
Note: The red test is a known sporadic error.

**Pull request merged**